### PR TITLE
implement string.match(regexp)

### PR DESCRIPTION
**Pull request by muskuloes**
an implementation of strings.match from issue #114

### CODE REVIEW

**Comment by jasonwilliams** [comment]
Thanks, im just looking into the benchmarks issue so bear with me

**Comment by jasonwilliams** [comment]
I've made some changes to the benchmarks Action, could you try rebasing and pushing again?

**Comment by muskuloes** [comment]
The benchmarks still failed.

**Comment by jasonwilliams** [comment]
Ugh... 
bear with me, might be a couple of days

File: src/lib/js/string.rs [hunk 1]
```diff
@@ -413,6 +415,81 @@ pub fn last_index_of(this: &Value, args: &[Value], ctx: &mut Interpreter) -> Res
...
+    let mut regex_body = String::new();
+    let mut regex_flags = String::new();
+    match args[0].deref() {
+        ValueData::String(ref body) => {
+            // first argument is a string -> use it as regex pattern
+            regex_body = body.into();
+        }
+        ValueData::Object(ref obj) => {
+            let slots = &*obj.borrow().internal_slots;
+            if slots.get("RegExpMatcher").is_some() {
```
Comment (jasonwilliams) [inline src/lib/js/string.rs]: You could use .get_internal_slot() here, which gives you either the value or `undefined`

Comment (jasonwilliams) [reply src/lib/js/string.rs]: i think you also want to handle the case where RegExpMatcher isn't there.
(An object was passed but it wasn't a RegExp)

File: src/lib/js/string.rs [hunk 2]
```diff
@@ -413,6 +415,81 @@ pub fn last_index_of(this: &Value, args: &[Value], ctx: &mut Interpreter) -> Res
...
+                // first argument is another `RegExp` object, so copy its pattern and flags
+                if let Some(body) = slots.get("OriginalSource") {
+                    regex_body = from_value(body.clone()).unwrap();
+                }
+                if let Some(flags) = slots.get("OriginalFlags") {
+                    regex_flags = from_value(flags.clone()).unwrap();
+                }
+            }
```
Comment (jasonwilliams) [inline src/lib/js/string.rs]: You could probably just return undefined here if RegExpMatcher isn't there

**Comment by muskuloes** [comment]
Hi, should I move the core functionality to `regexp.rs` as per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@match as well?

**Comment by jasonwilliams** [comment]
Yeap, although we don't have support for Symbols yet

**Comment by muskuloes** [comment]
Yes. I will. I'll push the update over the weekend.

**Comment by muskuloes** [comment]
PR updated

**Comment by jasonwilliams** [comment]
This is really awesome, thanks so much, looks clean and the tests are great

**Pull request merged**